From 600cedb14583767f3eb2dddcd78dfc86e562e8e6 Mon Sep 17 00:00:00 2001
From: Paul Keith <javelinanddart@gmail.com>
Date: Fri, 27 Jul 2018 03:02:49 +0200
Subject: [PATCH 6/7] sdk: Make smarter low-power display mode choices

* Keep CABC in low-power mode, it helps save battery
* Keep ACO in low-power mode, it helps save battery
* Only disable color enhancement if it's not the default,
  because some devices *need* color enhancement enabled

Change-Id: Id8c47bdf06782d783e175cd679847c227f3b3636
---
 .../internal/display/DisplayHardwareController.java       | 8 +++-----
 1 file changed, 3 insertions(+), 5 deletions(-)

diff --git a/lineage/lib/main/java/org/lineageos/platform/internal/display/DisplayHardwareController.java b/lineage/lib/main/java/org/lineageos/platform/internal/display/DisplayHardwareController.java
index eff3fde..85fe9a3 100644
--- a/lineage/lib/main/java/org/lineageos/platform/internal/display/DisplayHardwareController.java
+++ b/lineage/lib/main/java/org/lineageos/platform/internal/display/DisplayHardwareController.java
@@ -231,8 +231,7 @@ public class DisplayHardwareController extends LiveDisplayFeature {
         if (!mUseAutoContrast) {
             return;
         }
-        mHardware.set(LineageHardwareManager.FEATURE_AUTO_CONTRAST,
-                !isLowPowerMode() && isAutoContrastEnabled());
+        mHardware.set(LineageHardwareManager.FEATURE_AUTO_CONTRAST, isAutoContrastEnabled());
     }
 
     /**
@@ -243,7 +242,7 @@ public class DisplayHardwareController extends LiveDisplayFeature {
             return;
         }
         mHardware.set(LineageHardwareManager.FEATURE_COLOR_ENHANCEMENT,
-                !isLowPowerMode() && isColorEnhancementEnabled());
+                (!isLowPowerMode() || mDefaultColorEnhancement) && isColorEnhancementEnabled());
     }
 
     /**
@@ -253,8 +252,7 @@ public class DisplayHardwareController extends LiveDisplayFeature {
         if (!mUseCABC) {
             return;
         }
-        mHardware.set(LineageHardwareManager.FEATURE_ADAPTIVE_BACKLIGHT,
-                !isLowPowerMode() && isCABCEnabled());
+        mHardware.set(LineageHardwareManager.FEATURE_ADAPTIVE_BACKLIGHT, isCABCEnabled());
     }
 
     private synchronized void updateColorAdjustment() {
-- 
2.17.1

